### PR TITLE
feat: enhance code performance

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
the PR is to enhance the code performance by not doing N square time complexity loop, instead using `set` data structure to improve to O(1) time complexity

e.g:
- in deserialize method, when execute filter method, it compare if the filter condition in accountDetails by using `Object.keys(this.accountDetails).includes(XXX)` , which is a O(n) time complexity, and Object.keys will create extra memory for every filter loop
  - instead, we can use using set to reduce the complexity
- the same case as `migrateAccountDetails` method
- in removeAccount method, it is using a redundant loop to search where the address is exist or not

### CODE REVIEW

File: src/ledger-keyring.ts [hunk 1]
```diff
@@ -157,26 +156,25 @@ export class LedgerKeyring extends EventEmitter {
...
+          if (!keys.has(key)) {
+            this.accountDetails[key] = {
               bip44: false,
               hdPath: this.#pathFromAddress(account),
             };
-          } catch (error) {
-            console.log(`failed to migrate account ${account}`);
           }
-        });
+          // fix a edge case when accounts itself has duplicate address
```
[the remark this answers is not in the document]
Comment (stanleyyconsensys) [reply src/ledger-keyring.ts]: > It looks to me like this part is actually a fix unrelated to the performance improvement - does it warrant breaking out and a test-case? Also: Do we know that this is always fine?

actually i find that is no necessary, or it will not cause any issue if any edge case happen
e.g
accounts = [acc1, acc1, acc2] , accountDetails = {acc2}

if we take a dry run on the Prev code
1. filter accounts where those not in accountDetails, accounts become  [acc1, acc1]
2. acc1 not in accountDetails, acc1 add to accountDetails 
3. acc1 not in accountDetails, acc1 add to accountDetails , but the content will be same
4. at the end result will become accountDetails = {acc2, acc1}

if we take a dry run on the Updated code
1. acc2 in set, skip
2. acc1 not in set, acc1 add to accountDetails , add to set
3. acc1 in set, skip
4. at the end result will become accountDetails = {acc2, acc1}

**Comment by stanleyyconsensys** [comment]
@legobeat can you merge for me, i dont have the permission
